sharding-core/sharding-core-rewrite/src/main/java/org/apache/shardingsphere/core/rewrite/feature/encrypt/parameter/impl/EncryptInsertValueParameterRewriter.java

```diff
@@ -57,10 +60,14 @@ public void rewrite(final ParameterBuilder parameterBuilder, final SQLStatementC
         if (!encryptTable.isPresent()) {
             return;
         }
-        for (String each : encryptTable.get().getLogicColumns()) {
-            Optional<ShardingEncryptor> shardingEncryptor = encryptRule.findShardingEncryptor(tableName, each);
+        Optional<InsertColumnsSegment> insertColumnsSegment = sqlStatementContext.getSqlStatement().findSQLSegment(InsertColumnsSegment.class);
+        if (!insertColumnsSegment.isPresent()) {
+            return;
+        }
+        for (ColumnSegment each : insertColumnsSegment.get().getColumns()) {
```

Comment:
Yes, I also forget add the test cases for insert without columns。
my purpose is:
in order to use the original order in SQL to rerwite cipher SQL, rather than using the column order in the configured encryption rules, otherwise it will also cause the parameters to be disordered.